Fixes #1327 - Copy image

fleur101 opened the pull request:
Fixes #1327 

- Adds handing not only base64 images
- Adds support for jpeg images in addition to png
- Fixes finding the biggest visible image 


Before submitting a final PR, please:

- [x] Run `npm test` on your machine
- [x] Run `npm run format` to keep code formatted consistently
- [x] Reference the issue you are fixing in at least one of your commit messages, as `Fixes #X`

ianb commented on extension/intents/clipboard/contentScript.js:
```diff
@@ -108,14 +108,17 @@ this.contentScript = (function() {
-        imgHeight.push(element.clientHeight);
+    let maxHeightIndex = 0;
+    const isValidSrc = src =>
+      src.startsWith("data:") || /png|jpe?g/.test(src.split(".").pop());
```
URLs can be anything, it's best not to check the text of the URL to try to determine the type of the file. Anything in an `<img>` should be an image of some sort.

fleur101 commented:
@ianb I am sorry, I am a bit confused when you say you do not understand the purpose of change. This pull request tries to fix the problem that the image is not copied or the wrong image is copied - I referenced it in the commit messages and now in the pull request description as well. 

There were two bugs.

First is when image src was not base64 encoded image. When I looked into the code, I found out that the code was dealing only with base64 images, in particular, the line ` const binary = atob(url.split(",", 2)[1]);` in `dataUrlToBlob` function of `clipboard.js` file returned _undefined_ for images whose source did not start with "data:image/jpeg;base64," or a similar pattern. I then found how to do that for other than base64 images here: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/clipboard/setImageData

The second bug was when selecting the biggest visible image. The wrong image was returned because the index in the `imgHeight ` array did not correspond to the index in `img `array since they have different sizes. I fixed the calculation.

As I understand now, I cannot rely on the url to get the image type.  I'll look into what you proposed. Thank you

fleur101 commented:
Currently, the copying function works the following way: 
It searches for the biggest (largest height) image that is visible on the screen. Then it tries to copy the image to the clipboard either by fetching the image from the url or by using "data:" url. `browser.clipboard.setImageData(imageData, imageType` supports only jpeg/png images, so if the biggest image on the screen has any other type (gif, bmp, etc.), then the image is not copied.

This PR fixes the following parts of this process:
- Search now finds the correct image
- Copy now supports fetching images from non-"data:" type src
- Copy now supports JPEG images in addition to PNG

ianb commented:
Thanks, these look like good improvements
